LFS-840: Add ability to delete pedigree

veronikaslc opened the pull request:
Now really saving pedigree data to the form upon clicking so in the pedigree editor.

Commit:
LFS-840: Add ability to delete pedigree

sashaandjic commented:
I am getting this error when saving in pedigree editor:
![image](https://user-images.githubusercontent.com/23243244/106559180-05ad3500-64f3-11eb-9407-365a8c7b0d80.png)

Commit:
LFS-840: Add ability to delete pedigree

veronikaslc commented:
@sashaandjic Fixed

sashaandjic commented:
One small anomaly found.  Steps to reproduce:
- create a new demographic form
- add a pedigree
- save the form - notice that no delete button has been added next to the pedigree
- refresh - the delete button appears
![image](https://user-images.githubusercontent.com/23243244/106644198-c7e2f780-6558-11eb-92ab-20b2dfb372e6.png)

Commit:
LFS-840: Add ability to delete pedigree

veronikaslc commented:
@sashaandjic Fixed

OrdiNeu committed:
LFS-840: Add ability to delete pedigree

Remove the onSave prop and use the formReader context.
Prevent an infinite loop by moving the pedigreeData to a state variable.

veronikaslc commented:
@sashaandjic Ready for retest

veronikaslc commented:
Resolved conflicts with dev.

veronikaslc commented:
Fixed some typos as well.

sashaandjic reviewed (commented):
tested - all good
![image](https://user-images.githubusercontent.com/23243244/106817433-dad2f600-6644-11eb-8d4d-e63bdf8344a3.png)
